openquake/hazardlib/gsim/parker_2020.py

```diff
@@ -0,0 +1,752 @@
+
+        stddevs = []
+
+        # total Phi
+        phi_rv = np.zeros(len(vs30))
+        for i, vs30i in enumerate(vs30):
+            if rrup[i] <= r1:
+                phi_rv[i] = C["phi21"]
+            elif rrup[i] >= r2:
+                phi_rv[i] = C["phi22"]
+            else:
+                phi_rv[i] = ((C["phi22"] - C["phi21"])
+                             / (math.log(r2) - math.log(r1))) \
+                            * (math.log(rrup[i]) - math.log(r1)) + C["phi21"]
+
+            if vs30i <= v1:
+                phi_rv[i] += C["phi2V"] * \
+                             (math.log(r2 / max(r1, min(r2, rrup[i])))
+                              / math.log(r2 / r1))
+            elif vs30i < v2:
+                phi_rv[i] += C["phi2V"] * \
+                          ((math.log(v2 / min(v2, vs30i)))
+                           / math.log(v2 / v1)) * \
+                          (math.log(r2 / max(r1, min(r2, rrup[i])))
+                           / math.log(r2 / r1))
+
+        phi_tot = np.sqrt(phi_rv)
+
+        for stddev in stddev_types:
+            assert stddev in self.DEFINED_FOR_STANDARD_DEVIATION_TYPES
+            if stddev == const.StdDev.TOTAL:
+                stddevs.append(np.sqrt(C["Tau"] ** 2 + phi_tot ** 2))
+            elif stddev == const.StdDev.INTER_EVENT:
+                stddevs.append(np.repeat(C["Tau"], len(vs30)))
+            elif stddev == const.StdDev.INTRA_EVENT:
+                stddevs.append(phi_tot)
+
+        return stddevs
+
+    def hypocentral_depth_estimate(self, ztor, width, dip):
+        """
+        Estimate hypocentral_depth if not available.
+        Ch. 4.3.3/Eqs. 4.16 & 4.17  of Parker et al. (2020).
+        """
+        return ztor + 0.48 * width * math.sin(dip)
+
+    def _c0(self, C, C_PGA):
+        """
+        c0 factor.
+        """
+        if self.saturation_region is None:
+            c0_col = "c0"
+        else:
+            c0_col = self.saturation_region + "_c0"
+        return C[c0_col], C_PGA[c0_col]
+
+    def _magnitude_scaling(self, C, C_PGA, mag, m_b):
+        """
+        Magnitude scaling factor.
+        """
+        m_diff = mag - m_b
+        sfx = self._suffix()
+        if m_diff > 0:
+            fm = C["c6" + sfx] * m_diff
+            fm_pga = C_PGA["c6" + sfx] * m_diff
+        else:
+            fm = C["c4" + sfx] * m_diff + C["c5" + sfx] * m_diff ** 2
+            fm_pga = C_PGA["c4" + sfx] * m_diff \
+                + C_PGA["c5" + sfx] * m_diff ** 2
+
+        return fm, fm_pga
+
+    def _get_mb(self):
+        """
+        Regional Mb factor.
+        """
+        mb_regions = ("Aleutian", "AK", "Cascadia", "CAM_S", "CAM_N",
+                      "JP_Pac", "JP_Phi", "SA_N", "SA_S", "TW_W", "TW_E")
+        if self.saturation_region in mb_regions:
+            i = mb_regions.index(self.saturation_region)
+            m_b = (8, 8.6, 7.7, 7.4, 7.4, 8.5, 7.7, 8.5, 8.6, 7.1, 7.1)[i]
+        else:
+            m_b = 7.9
+
+        return m_b
```

Comment:
Wouldn't a dictionary be cleaner here, i.e.:
```python
mb_regions = {"Aleutian": 8.0, "AK": 8.6, ... "TW_E": 7.1}
if self.saturation_region in mb_regions:
    return  mb_regions[self.saturation_region]
else:
    return 7.9
```
It might even be better still to make the dictionary `MB_REGIONS` an attribute of the class